### ISSUE TITLE
Polymorphic serialization: Multiple Property to define Sub Class

### ISSUE THREAD

**Issue by spremthi57** [FasterXML/jackson-databind#2957]
**Is your feature request related to a problem? Please describe.**
A clear and concise description of what the problem is.
Currently JsonTypeInfo property accepts only String, meaning only one property  we can pass to define Subclaseses. I am not able to Deserialize if my subclasses depends on two or more properties

**Describe the solution you'd like**
A clear and concise description of what you want to happen
JsonTypeInfo property accepts String Array and all the properties associated should be of type String. And properties associated should be of default as empty string

**Usage example**
If you have a clear idea of how to use proposed new/modified feature, please show an example.
public abstract class Animal {

     private String type;
     private String breed;

}

public class Dog extends Animal {
      private String dogCharacters;
}

public class Bulldog extends Dog {
       private boolean lessNoisy;

}

public class Cat extends Animal {

       private String catCharacters;
}

In this Dog and Cat, depends on type and Bulldog depends on type and breed

**Additional context**
Add any other context about the feature request here.

**Comment by cowtowncoder** [comment]
Right, there is no support for multi-property (combo) type ids, and no plans to add any.

**Comment by Ignasimg** [comment]
> Right, there is no support for multi-property (combo) type ids, and no plans to add any.

It's been a few years but given your recent closure of the issue - could you elaborate on the rationale behind this?

**Comment by cowtowncoder** [comment]
I do not see the need nor have interest in more complicated system over single-property, flat resolution that works for most cases.

If someone wants to work on implementation they are free to do a PR and I will consider it; but having issue open with no plans is pointless, with 200+ other issues open.
